### PR TITLE
statsV2: implement Failcnt

### PR DESCRIPTION
**Pull request by AkihiroSuda**
**- What I did**

Implemented `MemoryStats.Failcnt` for cgroup v2

**- How I did it**

Used the `oom` field of `memory.events` file

**- How to verify it**

```console
$ docker run -it --rm --name foo --memory 8M alpine
/ # dd if=/dev/urandom of=/dev/null bs=32M count=1
Killed
/ # dd if=/dev/urandom of=/dev/null bs=32M count=1
Killed
/ # cat /sys/fs/cgroup/memory.events
low 0
high 0
max 23
oom 2
oom_kill 2
```

```console
$ curl -s --unix-socket /var/run/docker.sock http://localhost/containers/foo/stats?stream=false | jq .memory_stats.failcnt
2
```

**- A picture of a cute animal (not mandatory but encouraged)**
:penguin:

### CODE REVIEW

File: daemon/daemon_unix.go [hunk 1]
```diff
@@ -1560,12 +1560,15 @@ func (daemon *Daemon) statsV2(s *types.StatsJSON, stats *statsV2.Metrics) (*type
...
 			// MaxUsage is not supported
 			Limit: stats.Memory.UsageLimit,
-			// TODO: Failcnt
 		}
 		// if the container does not set memory limit, use the machineMemory
 		if s.MemoryStats.Limit > daemon.machineMemory && daemon.machineMemory > 0 {
 			s.MemoryStats.Limit = daemon.machineMemory
 		}
+		if stats.MemoryEvents != nil {
+			s.MemoryStats.Failcnt = stats.MemoryEvents.Oom
```
Comment (thaJeztah) [inline daemon/daemon_unix.go:1569]: Do you think this requires a comment to explain? It may not be obvious how these relate

**Review by thaJeztah — COMMENTED** [review]
Would it be ok to do the vendor update separate or in a separate commit? I think some of the changes were related to compatibility with Go 1.14, and depending on how long 19.03 is still "a thing" perhaps we should backport those (not sure if possible)

File: daemon/daemon_unix.go (hunk 1 above)
Comment (AkihiroSuda) [reply daemon/daemon_unix.go:1571]: updated

**Comment by AkihiroSuda** [comment]
updated to split the commits

**Review by thaJeztah — APPROVED** [review]
LGTM

**Comment by AkihiroSuda** [comment]
@kolyshkin PTAL?

**Comment by kolyshkin** [comment]
I'm borderline as to whether it makes sense to add an integration test for it or now.

Otherwise LGTM

**Review by kolyshkin — APPROVED** [review]
I'm borderline as to whether it makes sense to add an integration test for it or now.

Otherwise LGTM

**Comment by kolyshkin** [comment]
> I'm borderline as to whether it makes sense to add an integration test for it or now.

Or is there such a test already? Should be pretty easy to implement

**Comment by AkihiroSuda** [comment]
We don't have the test even for cgroup v1. I guess it can be another PR.

**Review by tao12345666333 — APPROVED** [review]

**Comment by tao12345666333** [comment]
need rebase.

**Comment by AkihiroSuda** [comment]
rebased

**Review by thaJeztah — APPROVED** [review]
ok let's merge this one (and look for more tests in a follow up)

LGTM

**Pull request merged**